ipa-core/src/ff/boolean_array.rs

```diff
@@ -138,7 +138,7 @@ macro_rules! boolean_array_impl {
                     let v = v.into();
                     let mut val = Self::ZERO;
                     for i in 0..$bits {
-                        val.set(i, Boolean::from((v >> i & 1) == 1));
+                        val.set(i, Boolean::from(i < 128 && (v >> i & 1) == 1));
```

Comment:
can you put a comment explaining why and when we are going to fix it here?

Reply:
all right, I think the semantic is not changed and we accept only 128 bit values. there is no reason to iterate over all of them, you can just take first 128 or build a mask and do a single AND instruction - compiler likely can do it too, but I haven't looked at assembly code, so I can't be sure